### PR TITLE
drivers: kscan: mchp: low power mode enabled

### PR DESCRIPTION
**Pull request by Manimaran-A**
Updated KSCAN driver and dts files to support low power feature.

### CODE REVIEW

**Comment by Manimaran-A** [comment]
@albertofloyd could you please help to review?

**Comment by MaureenHelm** [comment]
@jvasanth1 @albertofloyd will you please take a look?

File: drivers/kscan/kscan_mchp_xec.c [hunk 1]
```diff
@@ -411,6 +418,9 @@ void polling_task(const struct device *dev, void *dummy2, void *dummy3)
 			/* Allow other threads to run while we sleep */
 			k_usleep(wait_period);
 		}
+		#ifdef CONFIG_PM_DEVICE
```
Comment (albertofloyd) [inline drivers/kscan/kscan_mchp_xec.c:421]: nit, do we need to space ifdef like this?

**Review by albertofloyd — DISMISSED** [review]

**Review by jvasanth1 — DISMISSED** [review]

**Comment by fabiobaltieri** [comment]
@Manimaran-A could you rebase please?

File: drivers/kscan/kscan_mchp_xec.c (hunk 1 above)
Comment (Manimaran-A) [reply drivers/kscan/kscan_mchp_xec.c]: Ok, removed the space before #ifdef.

**Comment by Manimaran-A** [comment]
@albertofloyd @jvasanth1 Could you please re-approve the PR

**Review by jvasanth1 — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**